src/pks/transport/transport_ats_pk.cc

```diff
@@ -52,6 +53,17 @@ Transport_ATS::Transport_ATS(Teuchos::ParameterList& pk_tree,
   PK(pk_tree, glist, S, solution),
   PK_PhysicalExplicit<Epetra_Vector>(pk_tree, glist, S, solution)
 {
+
+
+  key_ = Keys::readKey(*plist_, domain_, "primary variable", "total_component_concentration");
+  tcc_key_ = key_;
+  
+  // set up the primary variable solution, and its evaluator
+  Teuchos::ParameterList& FElist = S->FEList();
```

Comment:
Instead of doing this way, please use:

```
Teuchos::ParameterList& pv_sublist = S->GetEvaluatorList(key_);
```
https://github.com/amanzi/amanzi/blob/master/src/state/State.hh#L299

This is more robust because it allows for domain set names, which we use extensively in transport for subgrid models.